safenode/src/tests_e2e/registers_with_churn.rs

```diff
@@ -0,0 +1,253 @@
+                    let _ = regs_erred
+                        .write()
+                        .await
+                        .entry(addr)
+                        .and_modify(|curr| curr.attempts += 1)
+                        .or_insert(RegError {
+                            addr,
+                            attempts: 1,
+                            last_err,
+                        });
+                }
+            }
+        }
+    });
+}
+
+// Spawns a task which periodically picks up a random node, and restarts it to cause churn in the network.
+fn churn_nodes_task() {
+    let _handle = tokio::spawn(async {
+        let mut addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 12000);
+        let delay = Duration::from_millis(CHURN_PERIOD_MILLIS);
+        loop {
+            // let's choose a random node to restart
+            let node_index = rand::thread_rng().gen_range(1..=NODE_COUNT);
+            addr.set_port(12000 + node_index);
+
+            println!("Restarting node through its RPC service at {addr} in {delay:?}");
+            sleep(delay).await;
+
+            node_restart(addr, 1000)
+                .await
+                .expect("Failed to restart node with RPC endpoint {addr}");
+        }
+    });
+}
+
+// Checks (periodically) for any registers that an error was reported either at the moment of its creation or
+// in a later query attempt.
+fn retry_query_registers_task(
+    client: Client,
+    regs_erred: RegistersErredList,
+    failures: RegistersErredList,
+) {
+    let _handle = tokio::spawn(async move {
+        let delay = Duration::from_millis(2 * CHURN_PERIOD_MILLIS);
+        loop {
+            sleep(delay).await;
+
+            // let's try to query from the bucket of those that erred upon creation/query
+            let erred = regs_erred.write().await.pop_first();
+
+            if let Some((addr, mut reg_error)) = erred {
+                let attempts = reg_error.attempts + 1;
+                println!("Querying erred Register at {addr:?}, attempt: #{attempts} ...");
+                if let Err(last_err) = client.get_register(*addr.name(), addr.tag()).await {
+                    println!("Erred Register is still not retrievable at {addr:?} after {attempts} attempts: {last_err:?}");
+                    // We only keep it to retry 'MAX_NUM_OF_QUERY_ATTEMPTS' times,
+                    // otherwise report it effectivelly as failure.
+                    reg_error.attempts = attempts;
+                    reg_error.last_err = last_err;
+
+                    if attempts == MAX_NUM_OF_QUERY_ATTEMPTS {
+                        let _ = failures.write().await.insert(addr, reg_error);
+                    } else {
+                        let _ = regs_erred.write().await.insert(addr, reg_error);
+                    }
+                }
+            }
+        }
+    });
+}
+
+async fn node_restart(addr: SocketAddr, delay_millis: u64) -> Result<()> {
```

Comment:
Registers are read/write from disk on request.
Which actually a `disk-based-storage` requiring a targeted replication.

Meanwhile, a simple node restart won't clean out the previous disk files, which leaves a `back hole` for register to be read back during the churning test.
To resolve this:
1, Ensure during node restart, previous disk files will be cleaned up.
OR
2, instead of `node_restart` , have to `stop a node via rpc` then `add back a new node`.

Reply:
Pushed a commit implementing #1.